python/mcap-ros2-support/mcap_ros2/dynamic.py

```diff
@@ -0,0 +1,354 @@
+    return parser.uint32()
+
+
+def parseInt64(reader: CdrReader) -> int:
+    return reader.int64()
+
+
+def parseUint64(reader: CdrReader) -> int:
+    return reader.uint64()
+
+
+def parseString(reader: CdrReader) -> str:
+    return reader.string()
+
+
+def parseWstring(reader: CdrReader) -> str:
+    raise NotImplementedError("wstring parsing is not implemented")
+
+
+def parseTime(reader: CdrReader) -> Time:
+    seconds = reader.int32()
+    nanoseconds = reader.uint32()
+    return Time(seconds=seconds, nanoseconds=nanoseconds)
+
+
+def parseBoolArray(reader: CdrReader, array_length: int) -> List[bool]:
+    return [parseBool(reader) for _ in range(array_length)]
+
+
+def parseInt8Array(reader: CdrReader, array_length: int) -> List[int]:
+    return [reader.int8() for _ in range(array_length)]
+
+
+def parseUint8Array(reader: CdrReader, array_length: int) -> bytes:
+    return reader.uint8_array(array_length)
+
+
+def parseInt16Array(reader: CdrReader, array_length: int) -> List[int]:
+    return [reader.int16() for _ in range(array_length)]
+
+
+def parseUint16Array(reader: CdrReader, array_length: int) -> List[int]:
+    return [reader.uint16() for _ in range(array_length)]
+
+
+def parseInt32Array(reader: CdrReader, array_length: int) -> List[int]:
+    return [reader.int32() for _ in range(array_length)]
+
+
+def parseUint32Array(reader: CdrReader, array_length: int) -> List[int]:
+    return [reader.uint32() for _ in range(array_length)]
+
+
+def parseInt64Array(reader: CdrReader, array_length: int) -> List[int]:
+    return [reader.int64() for _ in range(array_length)]
+
+
+def parseUint64Array(reader: CdrReader, array_length: int) -> List[int]:
+    return [reader.uint64() for _ in range(array_length)]
+
+
+def parseFloat32Array(reader: CdrReader, array_length: int) -> List[float]:
+    return [reader.float32() for _ in range(array_length)]
+
+
+def parseFloat64Array(reader: CdrReader, array_length: int) -> List[float]:
+    return [reader.float64() for _ in range(array_length)]
+
+
+def parseStringArray(reader: CdrReader, array_length: int) -> List[str]:
+    return [reader.string() for _ in range(array_length)]
+
+
+def parseWstringArray(reader: CdrReader, array_length: int) -> List[str]:
+    raise NotImplementedError("wstring[] parsing is not implemented")
+
+
+def parseTimeArray(reader: CdrReader, array_length: int) -> List[Time]:
+    return [parseTime(reader) for _ in range(array_length)]
+
+
+FIELD_PARSERS = {
+    "bool": parseBool,
+    "byte": parseUint8,
+    "char": parseInt8,
+    "float32": parseFloat32,
+    "float64": parseFloat64,
+    "int8": parseInt8,
+    "uint8": parseUint8,
+    "int16": parseInt16,
+    "uint16": parseUint16,
+    "int32": parseInt32,
+    "uint32": parseUint32,
+    "int64": parseInt64,
+    "uint64": parseUint64,
+    "string": parseString,
+    "wstring": parseWstring,
+    "duration": parseTime,
+    "time": parseTime,
+}
+
+ARRAY_PARSERS = {
+    "bool": parseBoolArray,
```

Comment:
could use a functor here?

```python
def array_parser(primitive_parse_func):
    def parser(reader: CdrReader, length: int):
        return [primitive_parse_func(reader) for _ in range(length)]
    return parser

ARRAY_PARSERS = {
    "bool": array_parser(CdrReader.bool),
    ....
}
```

Python type annotations are a PITA for generics like this unfortunately.